### DISCUSSION TITLE
New Demo: Image Processor

### DISCUSSION THREAD

**Discussion by Geliogabalus** [clientIO/joint#2388] [News]
Wanted to process your images in a **node-based manner**? Here's our **image processing** application using the full power of JointJS+ and the open source **OpenCV library**. In the [Image Processor](https://resources.jointjs.com/demos/rappid/apps/ImageProcessor/) application, you can manipulate images using **filters and trasform tools**.

[OpenCV](https://opencv.org/) allows you to use a wide range of operations on images for a variety of purposes, from computer vision to filtering and image combining. The application uses JavaScript version of the library. 

In the example, you can see how images are loaded and **processed using nodes**, proving that creating large-scale workflows is possible.

![image](https://github.com/clientIO/joint/assets/8438073/5c671f39-1815-4355-b02a-88a09d946fa2)
_By default, you see an example where you can create a token from several images for various TTRPGs._

Here is a brief overview of the app's features:
- **Upload** - you can upload an image via the Upload node using the context menu or url. Alternatively, you can drag and drop images from your computer directly into the application.
- **Download** - you can download images from the View nodes using the context menu.
- **Files** - you can save and load a workflow file using the drop-down menu on the toolbar. The files created have the extension ".imp".
- **Nodes** - click on the question mark symbol of a node to get more information about each node.

**Comment by marekhozak10** [comment]
Already in my Bookmarks! 👏
